openedx/core/djangoapps/content/block_structure/models.py

```diff
@@ -0,0 +1,104 @@
+"""
+Models used by the block structure framework.
+"""
+
+from django.core.files.base import ContentFile
+from django.db import models
+
+from model_utils.models import TimeStampedModel
+from openedx.core.djangoapps.xmodule_django.models import UsageKeyField
+from openedx.core.lib.block_structure.exceptions import BlockStructureNotFound
+
+
+def upload_to_path(bs_model, filename):  # pylint:disable=W0613
+    """
+    Returns path name to use for the given
+    BlockStructureModel instance.
+    """
+    return unicode(bs_model.data_usage_key)
+
+
+class BlockStructureModel(TimeStampedModel):
+    """
+    Model for storing Block Structure information.
+    """
+    class Meta(object):
+        db_table = 'block_structure'
```

Comment:
This will probably be included as part of the "how to we purge" decision, but don't we want some uniqueness constraints on this model?